### ISSUE TITLE
Could we only finetune prompt encoder?

### ISSUE THREAD

**Issue by panboshui** [mazurowski-lab/finetune-SAM#28]

**Comment by haoyudong-97** [comment]
Yes, you can do so by turning off the gradient requirement for image encoder and mask decoder. See example codes at https://github.com/mazurowski-lab/finetune-SAM/blob/e41f73287329ea68df3bfb153c867bdec4eb23b2/SingleGPU_train_finetune_noprompt.py#L62-L64
